pkg/ddc/alluxio/cache.go

```diff
@@ -48,7 +48,9 @@ func (e *AlluxioEngine) queryCacheStatus() (states cacheStates, err error) {
 
 	dataset, err := utils.GetDataset(e.Client, e.name, e.namespace)
 	if err != nil {
-		e.Log.Error(err, "Failed to get dataset when query cache status")
+		if utils.IgnoreNotFound(err) != nil {
+			e.Log.Error(err, "Failed to get dataset when query cache status")
+		}
```

Comment:
Can we skip the code after  `states = e.ParseReportSummary(summary)` here. I think it makes no use to shutdown an engine.

From my understanding, the `cacheState` contains three kinds of metrics related to cache:
- basic cache status (e.g. `cached`, `cacheCapacity`)
- `cached percentage` which must be derived from `cached` and `dataset.UFSTotal`
- metrics related to cache hits (e.g. `cacheHitStates`)

Actually we only need the metric `cached`(checking if equals to zero) when shutting  down the runtime. So I suggest if we can have lower-level functions to query these three kinds of cache-related metrics. With such functions, the `AlluxioEngine.cleanupCache()` may avoid handling error like "Dataset Not found".

WDYT @cheyang

Reply:
I can make this refactoring in a next PR if it looks good to you @cheyang

Reply:
Thanks, I agree with your suggestion. `cached` is what we need when shutting down the engine.